src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/MissingJavadocMethodCheck.java

```diff
@@ -430,11 +430,8 @@ private boolean isMissingJavadocAllowed(final DetailAST ast) {
      * @return True if this method or constructor doesn't need Javadoc.
      */
     private boolean isContentsAllowMissingJavadoc(DetailAST ast) {
-        return (ast.getType() == TokenTypes.METHOD_DEF
-                || ast.getType() == TokenTypes.CTOR_DEF
-                || ast.getType() == TokenTypes.COMPACT_CTOR_DEF)
-                && (getMethodsNumberOfLine(ast) <= minLineCount
-                    || AnnotationUtil.containsAnnotation(ast, allowedAnnotations));
+        return getMethodsNumberOfLine(ast) <= minLineCount
+                || AnnotationUtil.containsAnnotation(ast, allowedAnnotations);
```

Comment:
> In this all the ast.getType == ..  are use less because they are the only tokens which is accepted so any token is going to be one of them because this method is called by

I see `TokenTypes.ANNOTATION_FIELD_DEF` in the list of accepted tokens, maybe it is a filter for that. Please see https://stackoverflow.com/a/48729302/15412365 and verify the removal.

Reply:
good catch, such conditions were added at https://github.com/checkstyle/checkstyle/commit/8179a34d87881f480b1b26b44bafcbcc6607cec9

so we need to have javadoc on record - https://github.com/checkstyle/checkstyle/issues/8497

Reply:
Done.
Sorry for taking too quick a decision on this pr. I have made the decision with some simple annotations. instead of `ANNOTATION_FIELD_DEF`